.storybook/test-runner.ts

```diff
@@ -155,25 +155,30 @@ async function expectStoryToMatchSnapshot(
         await Promise.all(waitForSelector.map((selector) => page.waitForSelector(selector)))
     }
 
-    await page.waitForTimeout(400) // Wait for effects to finish
-
-    // Wait for all images to load
-    await page.waitForFunction(() =>
-        Array.from(document.querySelectorAll('img')).every((i: HTMLImageElement) => i.complete)
-    )
-
     // snapshot light theme
     await page.evaluate(() => {
         document.body.setAttribute('theme', 'light')
     })
 
+    // Wait for all images to load
+    await page.waitForFunction(() => Array.from(document.images).every((i: HTMLImageElement) => i.complete))
+    await waitForPageReady(page)
+    await page.waitForLoadState('networkidle')
+    await page.waitForTimeout(500)
+
     await check(page, context, browser, 'light', storyContext.parameters?.testOptions?.snapshotTargetSelector)
 
     // snapshot dark theme
     await page.evaluate(() => {
         document.body.setAttribute('theme', 'dark')
     })
 
+    // Wait for all images to load
+    await page.waitForFunction(() => Array.from(document.images).every((i: HTMLImageElement) => i.complete))
+    await waitForPageReady(page)
+    await page.waitForLoadState('networkidle')
+    await page.waitForTimeout(100)
+
```

Comment:
Is this purposely in here twice?

Reply:
Yeah, I thought that switching modes could cause like css background images to change or whatever, and if they weren't loaded, this might be an issue.